src/main/java/net/snowflake/client/core/arrow/StructuredTypeDateTimeConverter.java

```diff
@@ -0,0 +1,130 @@
+package net.snowflake.client.core.arrow;
+
+import static net.snowflake.client.jdbc.SnowflakeType.TIMESTAMP_LTZ;
+import static net.snowflake.client.jdbc.SnowflakeType.TIMESTAMP_NTZ;
+import static net.snowflake.client.jdbc.SnowflakeType.TIMESTAMP_TZ;
+
+import java.sql.Date;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.util.TimeZone;
+import net.snowflake.client.core.SFException;
+import net.snowflake.client.jdbc.ErrorCode;
+import net.snowflake.client.jdbc.SnowflakeType;
+import org.apache.arrow.vector.util.JsonStringHashMap;
+
+public class StructuredTypeDateTimeConverter {
+
+  private final TimeZone sessionTimeZone;
+  private final long resultVersion;
+  private final boolean honorClientTZForTimestampNTZ;
+  private final boolean treatNTZAsUTC;
+  private final boolean useSessionTimezone;
+  private final boolean formatDateWithTimeZone;
+
+  public StructuredTypeDateTimeConverter(
+      TimeZone sessionTimeZone,
+      long resultVersion,
+      boolean honorClientTZForTimestampNTZ,
+      boolean treatNTZAsUTC,
+      boolean useSessionTimezone,
+      boolean formatDateWithTimeZone) {
+
+    this.sessionTimeZone = sessionTimeZone;
+    this.resultVersion = resultVersion;
+    this.honorClientTZForTimestampNTZ = honorClientTZForTimestampNTZ;
+    this.treatNTZAsUTC = treatNTZAsUTC;
+    this.useSessionTimezone = useSessionTimezone;
+    this.formatDateWithTimeZone = formatDateWithTimeZone;
+  }
+
+  public Timestamp getTimestamp(
+      JsonStringHashMap<String, Object> obj, SnowflakeType type, TimeZone tz, int scale)
+      throws SFException {
+    if (tz == null) {
+      tz = TimeZone.getDefault();
+    }
+    switch (type) {
+      case TIMESTAMP_LTZ:
+        return convertTimestampLtz(obj, scale);
+      case TIMESTAMP_NTZ:
+        return convertTimestampNtz(obj, tz, scale);
+      case TIMESTAMP_TZ:
+        return convertTimestampTz(obj, scale);
+    }
+    throw new SFException(
+        ErrorCode.INVALID_VALUE_CONVERT,
+        "Unexpected Arrow Field for " + type.name() + " and object type " + obj.getClass());
+  }
+
+  public Date getDate(int value, TimeZone tz) throws SFException {
```

Comment:
Does int type is enough here?

Reply:
This is what is expected in Arrow converted as well, so I assumed that it is enough